<Type Name="DivideByZeroException" FullName="System.DivideByZeroException" FullNameSP="System_DivideByZeroException" Maintainer="ecma">
  <TypeSignature Language="ILASM" Value=".class public serializable DivideByZeroException extends System.ArithmeticException" />
  <TypeSignature Language="C#" Value="public class DivideByZeroException : ArithmeticException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit DivideByZeroException extends System.ArithmeticException" />
  <MemberOfLibrary>BCL</MemberOfLibrary>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyPublicKey>[00 00 00 00 00 00 00 00 04 00 00 00 00 00 00 00 ]</AssemblyPublicKey>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <ThreadingSafetyStatement>All public static members of this type are safe for multithreaded operations. No instance members are guaranteed to be thread safe.</ThreadingSafetyStatement>
  <Base>
    <BaseTypeName>System.ArithmeticException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <para> Represents the error that is caused by an attempt to divide a
      number by zero.</para>
    </summary>
    <remarks>
      <block subset="none" type="note">
        <para>The following CIL instructions throw <see cref="T:System.DivideByZeroException" />:</para>
        <list type="bullet">
          <item>
            <term>
         div</term>
          </item>
          <item>
            <term>
         div.un</term>
          </item>
          <item>
            <term>
         rem</term>
          </item>
          <item>
            <term>rem.un</term>
          </item>
        </list>
      </block>
    </remarks>
    <example>
      <para>The following example demonstrates an error that causes
      a <see cref="T:System.DivideByZeroException" /> exception.</para>
      <code lang="C#">using System;
public class DivideZeroTest {
 public static void Main() {
   int x = 0;
   try { 
     int y = 100/x;
   }
   catch (DivideByZeroException e) {
     Console.WriteLine("Error: {0}",e);
   }
 }
}
   </code>
      <para>The output is</para>
      <code>
Error: System.DivideByZeroException: Attempted to divide by zero.
   at DivideZeroTest.Main()
 </code>
    </example>
  </Docs>
  <Members>
    <Member MemberName=".ctor">
      <MemberSignature Language="ILASM" Value="public rtspecialname specialname instance void .ctor()" />
      <MemberSignature Language="C#" Value="public DivideByZeroException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue />
      <Parameters />
      <Docs>
        <summary>
          <para>Constructs and initializes a new instance of the <see cref="T:System.DivideByZeroException" /> class.</para>
        </summary>
        <remarks>
          <para>This constructor initializes the <see cref="!:System.DivideByZeroException.Message" /> property of the new instance to a system-supplied
   message that describes the error, such as "Attempted to divide by zero." This
   message takes into account the current system culture.</para>
          <para>The <see cref="!:System.DivideByZeroException.InnerException" /> property is initialized to <see langword="null" /> .</para>
        </remarks>
      </Docs>
      <Excluded>0</Excluded>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="ILASM" Value="public rtspecialname specialname instance void .ctor(string message)" />
      <MemberSignature Language="C#" Value="public DivideByZeroException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue />
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">A <see cref="T:System.String" /> that describes the error. The content of <paramref name="message" /> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture.</param>
        <summary>
          <para>Constructs and initializes a new instance of the <see cref="T:System.DivideByZeroException" />
class.</para>
        </summary>
        <remarks>
          <para>This constructor initializes the <see cref="!:System.DivideByZeroException.Message" /> property of the new instance using
<paramref name="message" /> . If <paramref name="message" /> is 
<see langword="null" />, the <see cref="!:System.DivideByZeroException.Message" /> property is initialized to the 
   system-supplied message provided by the constructor that takes no arguments. The <see cref="!:System.DivideByZeroException.InnerException" /> property is initialized to <see langword="null" /> .</para>
        </remarks>
      </Docs>
      <Excluded>0</Excluded>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected DivideByZeroException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">To be added.</param>
        <param name="context">To be added.</param>
        <summary>To be added.</summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="ILASM" Value="public rtspecialname specialname instance void .ctor(string message, class System.Exception innerException)" />
      <MemberSignature Language="C#" Value="public DivideByZeroException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue />
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">A <see cref="T:System.String" /> that describes the error. The content of <paramref name="message" /> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture.</param>
        <param name="innerException">An instance of <see cref="T:System.Exception" /> that is the cause of the current exception. If <paramref name="innerException" /> is non-null, then the current exception was raised in a catch block handling <paramref name="innerException" /> .</param>
        <summary>
          <para>Constructs and initializes a new instance of the <see cref="T:System.DivideByZeroException" />
class.</para>
        </summary>
        <remarks>
          <para>This constructor initializes the <see cref="!:System.DivideByZeroException.Message" /> property of the new instance using <paramref name="message" />, and
   the <see cref="!:System.DivideByZeroException.InnerException" /> property using <paramref name="innerException" />. If <paramref name="message" /> is
<see langword="null" />, the <see cref="!:System.DivideByZeroException.Message" /> property is initialized to the 
   system-supplied message provided by the constructor that takes no arguments.</para>
          <para>
            <block subset="none" type="note"> For information on inner exceptions,
   see <see cref="P:System.Exception.InnerException" />.</block>
          </para>
        </remarks>
      </Docs>
      <Excluded>0</Excluded>
    </Member>
  </Members>
  <TypeExcluded>0</TypeExcluded>
</Type>
